keyboards/palette1202/palette1202.c

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2019 niltea
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "palette1202.h"
+
+// set base layer and save it to EPROM
+void setBaseLayer(uint16_t layerID) {
+  eeconfig_update_default_layer(1UL<<layerID);
+  default_layer_set(1UL<<layerID);
+}
```

Comment:
This functionality already exists, actually: https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer